esp-hal/src/dma/pdma.rs

```diff
@@ -1005,3 +1062,127 @@ impl<'d> Dma<'d> {
         }
     }
 }
+
+#[doc(hidden)]
+pub enum AnyPdmaChannel {
+    Spi2(Spi2DmaChannel),
+    Spi3(Spi3DmaChannel),
+    I2s0(I2s0DmaChannel),
+    #[cfg(i2s1)]
+    I2s1(I2s1DmaChannel),
+}
```

Comment:
Just to re-iterate my two cents (but with some/this code as context this time), the type alias way would be less lines of code than the delegate way, would give PDMA chips type erasure for free and would skip the runtime check.

Reply:
It's pretty easy to rip out all this, we'll be able to do that at any point.